### PR TITLE
[Ready for Review] Better fix for NCCL + sccache

### PR DESCRIPTION
**Pull request by yf225**
`sccache` was stuck when compiling NCCL likely because `NUM_JOBS` was not set correctly. This PR tests this hypothesis.

### CODE REVIEW

**Comment by SsnL** [comment]
Just to bring this line to attention, not sure if it matters:
NCCL `CmakeList.txt`'s `make` command also sets `-j` at https://github.com/pytorch/pytorch/blob/master/third_party/nccl/CMakeLists.txt#L17 .

Also, I believe the original method of temporarily writing `NVCC_CUDA_EXECUTABLE` could be better done via `env CUDA_NVCC_EXECUTABLE=$(which nvcc)` before `${CMAKE_VERSION}` (still in a same line).

File: tools/build_pytorch_libs.sh [hunk 1]
```diff
@@ -217,19 +212,13 @@ function build_nccl() {
               -DCMAKE_CXX_FLAGS="$C_FLAGS $CPP_FLAGS $USER_CFLAGS" \
               -DCMAKE_SHARED_LINKER_FLAGS="$USER_LDFLAGS" \
```
Comment (SsnL) [inline tools/build_pytorch_libs.sh]: I believe you need to pass `NUM_JOBS` to `cmake` for `third_party/nccl/CMakeList.txt` to use it., e.g., `-DNUM_JOBS="$NUM_JOBS" \`

**Comment by yf225** [comment]
Thanks @SsnL! I believe this should be the right fix now.

**Review by ssnl — APPROVED** [review]
Looks legit!

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
cc: @orionr we modified third_party/nccl here

**Pull request merged**